### ISSUE TITLE
Create stalebot cleanup of issues?

### ISSUE THREAD

**Issue by stevenhorsman** [kata-containers/kata-containers#9459]
In [November](https://github.com/kata-containers/kata-containers/pull/8348) we added stalebot workflow to auto close PRs that had not action for 180 days (with a 7 day warning). That seems to have helped the PR backlog stop growing (IIRC we had just over 200 then and are at 184 as I type).

(As as aside we started the clean up in May 2023, so if we removed that then an extra ~63 PRs would be warned about closing.

We hit 1,000 issues in July 2023 and are now at 1.3k, so we should consider whether to do the same stalebot tidy up?

Thoughts?

**Comment by gkurz** [comment]
Quite often, the issue is created along the corresponding PR _just_  to pass CI, without any extra information in it. Could stalebot close an issue for which it already closed an associated PR, based on the `Fixes` stanza ?

**Comment by stevenhorsman** [comment]
> Quite often, the issue is created along the corresponding PR just to pass CI, without any extra information in it. Could stalebot close an issue for which it already closed an associated PR, based on the Fixes stanza ?

That should automatically be done by Github for us e.g. https://github.com/kata-containers/kata-containers/issues/9477 was closed this as completed in https://github.com/kata-containers/kata-containers/pull/9479 when Fabiano merged it.

Or are you talking about issues that map to PRs that were never merged?

**Comment by gkurz** [comment]
> > Quite often, the issue is created along the corresponding PR just to pass CI, without any extra information in it. Could stalebot close an issue for which it already closed an associated PR, based on the Fixes stanza ?
> 
> That should automatically be done by Github for us e.g. #9477 was closed this as completed in #9479 when Fabiano merged it.
> 
> Or are you talking about issues that map to PRs that were never merged?

Yeah the latter. Does stalebot also close issue XXX if stale PR had `Fixes XXXX` ?

**Comment by stevenhorsman** [comment]
This was discussion on the AC meeting 16th April. This is my summary, but other attendees should please keep me honest:
- Stale bot for issues is more tricky as new users cannot re-open issues, so if we implement this we need a longer warning period (e.g. at least 14 days rather than 7) and maybe allow issues to be left for 9 months over 180 days before being marked as stale. We should also request re-testing of bugs with the latest version as part of the stale message
- There is a concern that bugs should be handled differently in this process, (maybe we keep bugs back to the 3.0 release date?) but that required work to ensure that issues are triaged to have the bug label: There are 257 issues that currently would need this: https://github.com/kata-containers/kata-containers/issues?q=is%3Aissue+is%3Aopen+-label%3Abug+-label%3Aenhancement+-label%3Afeature
- Stale bot doesn't address the fundamental problem that we don't have the capacity/prioritisation to properly triage our issues, but it might help remove some noise
- Trying to understand how many issues were related to closed PRs via fixes, would be interesting.

**Comment by zvonkok** [comment]
We can add a github action for reopening an issue with  a command `/reopen` or something like that.

**Comment by stevenhorsman** [comment]
This stale issue clean up issue has become stale, and we didn't agree on how to clean up issues, so the problem is worse. How about we go for stale for one year and give a 30 day notice period. At the moment this would cover 13 release periods and give a user plenty of time to bump the issue if they still care about it?

**Comment by wainersm** [comment]
In the meeting today I suggested to get in touch the contributors active in the project, ask them to clean up stale issues. I was wondering how many issues we would be touching by that approach.

According to the following report, generate by Claude, top 20 submitters account for 42.4% of issues. Looking at the names in the table, there are many active members. Even for some not-active anymore you might still find them on your company and perhaps you could just drop a message like "hey, can I close your issues in your behalf for those that I think makes no more sense?" Would that approach be immoral or improper in some means? 

Not saying it will solve the issue, we still need a better long term approach,  but it can gives us an v0. 

```
Perfect! Here's the complete analysis of all 1,514 open issues by submitter:

  Top 20 Submitters:

  1. jodh-intel - 118 issues
  2. fidencio - 110 issues
  3. Apokleos - 43 issues
  4. wainersm - 41 issues
  5. studychao - 40 issues
  6. liubin - 35 issues
  7. c3d - 32 issues
  8. danmihai1 - 31 issues
  9. stevenhorsman - 28 issues
  10. zvonkok - 28 issues
  11. egernst - 25 issues
  12. dgibson - 19 issues
  13. YushuoEdge - 16 issues
  14. bergwolf - 16 issues
  15. burgerdev - 14 issues
  16. xyjixyjixyji - 14 issues
  17. Alex-Carter01 - 13 issues
  18. Tim-0731-Hzt - 13 issues
  19. fengwang666 - 12 issues
  20. gkurz - 12 issues

  Statistics:

  - Total open issues: 1,514
  - Unique submitters: 431
  - Top 10 submitters account for: 494 issues (32.6%)
  - Top 20 submitters account for: 642 issues (42.4%)
  - Submitters with only 1 issue: 305 (70.8% of all submitters)
  - Submitters with 10+ issues: 21 (4.9% of submitters, but 48.6% of issues)
```

**Comment by burgerdev** [comment]
Let me try to put some of my rambling from today's AC call in writing.

My experience with stalebot deployements is that they come off as rather user-hostile. I've seen many bug reports where a maintainer responded with "yes, this is a problem", which were then closed just because of age. It's even worse in the cases where only maintainers can remove the stale label, because the reason to add stalebot usually is maintainer overload, but overloaded maintainers don't usually get back to issues just because of a bot message.

What I'd prefer to see is some sort of triage system (I believe there are even labels for this right here). New bug reports and feature requests should be looked at by a maintainer, and if they meet some criteria should move to an "accepted" state. In that state, they should remain open until fixed, or positively identified as obsolete - never automatically closed (but maybe robot-nudged). For issues that are _not_ accepted, we can let the bot leave an apology like 

> Sorry, maintainers don't seem to have time, or maybe they missed this issue entirely. This will close in X days if not accepted. You can try asking on Slack if this was an oversight (or to get an idea why nobody responded).

Then we should probably monitor newly submitted issues and see that they are triaged - otherwise we'll just enter a loop of not triaged -> not accepted -> stale -> closed -> duplicate reported. And, frankly, I think without a plan for handling inbound issues in the future, we might as well not care about the existing issues.

**Comment by stevenhorsman** [comment]
I think the problem is that we don't have any scope to review all the issues (either the 1.5k with have, or every new issue). When we've discussed this in the past we talked about doing some triage at the end of community meetings if we have spare time, but we nearly always have too many topics to discuss.

**Comment by stevenhorsman** [comment]
The only thing we managed to agree on is that any issues raised pre Kata 3.0 (October 2022) could run through the stale bot process.

**Comment by burgerdev** [comment]
> I think the problem is that we don't have any scope to review all the issues (either the 1.5k with have, or **every new issue**).

(emphasis mine)

If we already know that we're not going to be able to review all _new_ issues, this is a separate problem with which stalebot does not help. As I mentioned in the meeting, if the goal is to only have valid issues and we assume we can't stem the flood of issues being opened, the only solution I can think of is to throttle the amount of issues being opened, somehow. The example I had in mind was https://www.github.com/ghostty-org/ghostty/issues/3558. I understand that this just redirects the flood elsewhere, but then I'm asking myself: What are we trying to accomplish by closing issues without review? 

* If we want an actionable list of issues, ghostty has a point.
* If we want to take better care of all reported issues, neither moving them nor closing them will help.
